libs/experimental/langchain_experimental/graph_transformers/llm.py

```diff
@@ -158,6 +159,97 @@ def map_to_base_relationship(rel: Any) -> Relationship:
     )
 
 
+def _parse_and_clean_json(
+    argument_json: Dict[str, Any],
+) -> Tuple[List[Node], List[Relationship]]:
+    nodes = []
+    for node in argument_json["nodes"]:
+        if not node.get("id"):  # Id is mandatory, skip this node
+            continue
+        nodes.append(
+            Node(
+                id=node["id"].title(),
+                type=node.get("type").capitalize() if node.get("type") else None,
+            )
+        )
+    relationships = []
+    for rel in argument_json["relationships"]:
+        # Mandatory props
+        if (
+            not rel.get("source_node_id")
+            or not rel.get("target_node_id")
+            or not rel.get("type")
+        ):
+            continue
+
+        # Node type copying if needed from node list
+        if not rel.get("source_node_type"):
+            try:
+                rel["source_node_type"] = [
+                    el["type"]
+                    for el in argument_json["nodes"]
+                    if el["id"] == rel["source_node_id"]
+                ][0].capitalize()
+            except IndexError:
+                rel["source_node_type"] = None
+        if not rel.get("target_node_type"):
+            try:
+                rel["target_node_type"] = [
+                    el["type"]
+                    for el in argument_json["nodes"]
+                    if el["id"] == rel["target_node_id"]
+                ][0].capitalize()
```

Comment:
can capitalize() fail with an error if list[0] == None ?

Reply:
same above

Reply:
it's going to be an index error in that case... and defaulting to None, but I'll change to safe access of types in case they are missing